### ISSUE TITLE
First run lint on just the changes in the PR before running it over the entire PR

### ISSUE THREAD

**Issue by ZainRizvi** [pytorch/pytorch#139201]
Goal: Usually lint errors are introduced by the dev's own changes. This can let the devs get a red signal on their PR significantly faster

**Comment by Skylion007** [comment]
I don't think this is significantly faster for anything aside from formatter linters and maybe flake8 linters? ruff is so fast it doesn't really matter. mypy needs to scan all files anyway and changes in the PR can cause type errors outside of it. For some linters like clang-format, isort, it could be faster, but not markedly so.
